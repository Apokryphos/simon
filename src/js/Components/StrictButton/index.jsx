import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function StrictButton(props) {
  const strictClass = classNames({
    'strict-button': true,
    enabled: props.enabled,
  });

  return (
    <div
      className="button-container"
      title="Toggles strict mode. In strict mode, the sequence resets if you press the wrong button."
    >
      <button className={strictClass} onClick={props.onClick} />
      <span className="button-label">Strict</span>
    </div>
  );
}

StrictButton.propTypes = {
  enabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default StrictButton;
