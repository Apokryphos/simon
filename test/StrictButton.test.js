import React from 'react';
import { shallow } from 'enzyme';
import StrictButton from 'Components/StrictButton';

describe('StrictButton rendering', () => {
  it('Renders OK', () => {
    const onClick = () => {};

    const wrapper = shallow(<StrictButton enabled onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props('enabled') === true);
    expect(wrapper.props('onClick') === onClick);
  });
});
