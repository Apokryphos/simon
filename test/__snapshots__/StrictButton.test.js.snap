// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StrictButton rendering Renders OK 1`] = `
<div
  className="button-container"
  title="Toggles strict mode. In strict mode, the sequence resets if you press the wrong button."
>
  <button
    className="strict-button enabled"
    onClick={[Function]}
  />
  <span
    className="button-label"
  >
    Strict
  </span>
</div>
`;
